libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -927,11 +955,20 @@ - (void)clearAccountState:(BOOL)clearAccountData {
         [self.coordinator.view removeFromSuperview];
     }
     
-    [SFAuthenticationManager removeAllCookies];
-    [self.coordinator stopAuthentication];
-
-    self.idCoordinator.idData = nil;
-    self.coordinator.credentials = nil;
+    __weak typeof(self) weakSelf = self;
+    [SFAuthenticationManager removeAllCookiesWithCompletion:^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [strongSelf.coordinator stopAuthentication];
```

Comment:
Having this here creates a race condition where authentication requests are canceled almost immediately.